add vtbwife

lianhong2758 opened the pull request.

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+			if err != nil {
+				panic(err)
+			}
+			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界
```
没用的删干净

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+			}*/
+			// 将文件内容转换为单词
+			keys = strings.Split(string(content), "\n")
+			log.Println("[vtbwife]加载", len(keys), "位wife数据...")
```
```go
log.Infoln("[vtbwife]加载", len(keys), "位 vtb ...")
```

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+		})).SetBlock(true).Handle(func(ctx *zero.Ctx) {
+		var key, u, b string
+		var ok bool
+		for i := 0; i < 3; i++ {
```
for i := 0; i < 3 && !ok; i++

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+		for i := 0; i < 3; i++ {
+			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i]
+			u, b, ok = geturl(key)
+			if !ok {
```
remove

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+			if !ok {
+				continue
+			}
+			break
```
remove

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+			break
+		}
+		if !ok {
+			ctx.SendChain(message.Text("-获取图片链接失败"))
```
这个诡异的`-`前缀是什么？全删了。

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+			ctx.SendChain(message.Text("-获取图片链接失败"))
+			return
+		}
+		img, err := web.GetData(u)
```
莲宝笨笨，直接把url放message.Image()不好吗？

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+	})
+	var k int
+	for kk, vv := range b {
+		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" {
```
你还真调用这么多次strings.TrimSpace啊🤣👉

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+		b = append(b, strings.TrimSpace(s.Text()))
+	})
+	var k int
+	for kk, vv := range b {
```
kk, vv：罚你重新起名😠

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+	doc.Find(".fans-medal-level").Remove()
+	var b []string
+	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) {
+		b = append(b, strings.TrimSpace(s.Text()))
```
有没有一种可能，就是说这里就是一个循环，你可以在这里把下面的事全干了

lianhong2758 replied:
要找开头,这个怎么在这里实现呢,如果没找到则全部写进去

lianhong2758 replied:
要找开头,这个怎么在这里实现呢,如果没找到则全部写进去

fumiama replied:
你下面怎么循环的，把它搬上来呗

fumiama commented on plugin/vtbwife/vtb.go:
```diff
@@ -0,0 +1,106 @@
+		}
+	}
+	if k != 0 {
+		brief = b[k-1] + "\n"
```
用strings.Builder

lianhong2758 commented:
TD




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月17日(星期一) 晚上8:11
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [FloatTech/ZeroBot-Plugin] add vtbwife (PR #672)





 
@fumiama requested changes on this pull request.
 
 
In go.mod:
 &gt; @@ -13,6 +13,7 @@ require (  	github.com/FloatTech/ttl v0.0.0-20220715042055-15612be72f5b  	github.com/FloatTech/zbpctrl v1.5.3-0.20230411055601-7cdff751dd67  	github.com/FloatTech/zbputils v1.6.2-0.20230411114057-4c71136244b2 +	github.com/PuerkitoBio/goquery v1.5.1  
莲宝快用已有的htmlquery
 
 
In plugin/vtbwife/vtb.go:
 &gt; + +func init() { // 插件主体 +	engine := control.Register("vtbwife", &amp;ctrl.Options[*zero.Ctx]{ +		DisableOnDefault: false, +		Brief:            "抽vtb老婆", +		Help:             "- 抽vtb(老婆)", +		PublicDataFolder: "VtbWife", +	}) +	var keys []string +	engine.OnRegex(`^抽(vtb|VTB)(老婆)?$`, fcext.DoOnceOnSuccess( +		func(ctx *zero.Ctx) bool { +			content, err := engine.GetLazyData("wife_list.txt", false) +			if err != nil { +				panic(err) +			} +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界  
没用的删干净
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	}) +	var keys []string +	engine.OnRegex(`^抽(vtb|VTB)(老婆)?$`, fcext.DoOnceOnSuccess( +		func(ctx *zero.Ctx) bool { +			content, err := engine.GetLazyData("wife_list.txt", false) +			if err != nil { +				panic(err) +			} +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界 +			if err != nil { +				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err) +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...")  log.Infoln("[vtbwife]加载", len(keys), "位 vtb ...")
 
 
In plugin/vtbwife/vtb.go:
 &gt; +			if err != nil { +				panic(err) +			} +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界 +			if err != nil { +				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err) +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ {  
for i := 0; i < 3 &amp;&amp; !ok; i++
 
 
In plugin/vtbwife/vtb.go:
 &gt; +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界 +			if err != nil { +				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err) +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok {  
remove
 
 
In plugin/vtbwife/vtb.go:
 &gt; +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok { +				continue +			} +			break  
remove
 
 
In plugin/vtbwife/vtb.go:
 &gt; +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok { +				continue +			} +			break +		} +		if !ok { +			ctx.SendChain(message.Text("-获取图片链接失败"))  
这个诡异的-前缀是什么？全删了。
 
 
In plugin/vtbwife/vtb.go:
 &gt; +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok { +				continue +			} +			break +		} +		if !ok { +			ctx.SendChain(message.Text("-获取图片链接失败")) +			return +		} +		img, err := web.GetData(u)  
莲宝笨笨，直接把url放message.Image()不好吗？
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	// 使用goquery解析网页内容 +	doc, err := goquery.NewDocumentFromReader(resp.Body) +	if err != nil { +		return "", "", false +	} +	u, ok = doc.Find(".infobox-image").Attr("src") // class加. +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text())) +	}) +	var k int +	for kk, vv := range b { +		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" {  
你还真调用这么多次strings.TrimSpace啊🤣👉
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	defer resp.Body.Close() +	// 使用goquery解析网页内容 +	doc, err := goquery.NewDocumentFromReader(resp.Body) +	if err != nil { +		return "", "", false +	} +	u, ok = doc.Find(".infobox-image").Attr("src") // class加. +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text())) +	}) +	var k int +	for kk, vv := range b {  
kk, vv：罚你重新起名😠
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	if err != nil { +		return "", "", false +	} +	defer resp.Body.Close() +	// 使用goquery解析网页内容 +	doc, err := goquery.NewDocumentFromReader(resp.Body) +	if err != nil { +		return "", "", false +	} +	u, ok = doc.Find(".infobox-image").Attr("src") // class加. +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text()))  
有没有一种可能，就是说这里就是一个循环，你可以在这里把下面的事全干了
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text())) +	}) +	var k int +	for kk, vv := range b { +		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" { +			k = kk + 1 +			break +		} +	} +	if k != 0 { +		brief = b[k-1] + "\n"  
用strings.Builder
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

lianhong2758 commented:
????




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月17日(星期一) 晚上8:23
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [FloatTech/ZeroBot-Plugin] add vtbwife (PR #672)





 
close
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

fumiama commented on go.mod:
```diff
@@ -13,6 +13,7 @@ require (
 	github.com/FloatTech/ttl v0.0.0-20220715042055-15612be72f5b
 	github.com/FloatTech/zbpctrl v1.5.3-0.20230411055601-7cdff751dd67
 	github.com/FloatTech/zbputils v1.6.2-0.20230411114057-4c71136244b2
+	github.com/PuerkitoBio/goquery v1.5.1
```
自己百度

guohuiyuan replied:
用gpt

lianhong2758 replied:
很稳定,两天测试没有问题,还可以统一数据

medicago087 commented:
莲宝该动一动了

lianhong2758 commented:
苜蓿子快帮我改改，求求了



---Original---
From: ***@***.***&gt;
Date: Fri, Apr 21, 2023 16:25 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [FloatTech/ZeroBot-Plugin] add vtbwife (PR #672)




 
莲宝该动一动了
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

medicago087 commented:
加油，我能力越小责任越小了

fumiama reviewed (changes requested):
有点气了😠你整扣扣米可以，谁让你搞这个了？zbpdata都合了，不用说不过去吧，

guohuiyuan commented:
这不直接dds攻击，搞挂它

lianhong2758 commented:
> 有点气了 😠 你整扣扣米可以，谁让你搞这个了？zbpdata都合了，不用说不过去吧，

呜呜呜
